00XX-common-ledger-protocol/00XX-common-ledger-protocol.md

```diff
@@ -0,0 +1,273 @@
+the `rejected` state, an `Error` should be returned.
+
+#### Expiry
+
+After the `expiresAt` date is reached, the transfer can no longer be fulfilled.
+If one party on the CLP connection is keeping authoritative state, they should
+send a `Reject` request to the other party. If both parties are keeping
+authoritative state, they can independently expire the transfer (set the state
+to `rejected` and roll it back) automatically.
+
+### Fulfill
+
+`Fulfill` is used to change an existing transfer from the `prepared` state to
+the `fulfilled` state. The packet-specific data of `Fulfill` is made up of `transferId`
+and `fulfillment`. `Fulfill` is a request with side effects: it changes the
+state of the transfer and finalizes the transfer's balance update. If the
+transfer is already fulfilled, no changes are applied.
+
+`transferId` is a 128-bit unique ID, which references an existing transfer.
+`fulfillment` is a 256-bit integer, containing the preimage of the
+`executionCondition` of the referenced transfer.
+
+#### Response
+
+A `Fulfill` request must be sent from the receiver of the referenced transfer,
+not the sender.
+
+A `Fulfill` request is successful if `transferId` references an existing
+transfer in the `prepared` state, the `fulfillment` is the SHA-256 preimage of
+the referenced transfer's `executionCondition`, and the current time is before
+the referenced transfer's `expiresAt`.
+
+A `Fulfill` request is also successful if the transfer is in the `fulfilled`
+state, and this packet matches the one that fulfilled the transfer,
+including side protocols but not including requestId.
+
+- `Response` is returned if the request is successful.  data to be passed back.
+  This means the transfer's state is now `fulfilled`, and its balance change
+has been finalized.
+
+- `Error` is returned if there is no transfer with the given ID, the
+  fulfillment does not match the condition, the expiresAt has already passed,
+or the transfer has been `rejected`.
+
+### Reject
+
+`Reject` is used to change an existing transfer from the `prepared`
+state to the `rejected` state. The packet-specific data of `Reject` is made up
+of `transferId` and `rejectionReason`.
+
+`transferId` is a 128-bit unique ID, which references an existing transfer.
+`rejectionReason` is an [ILP Error], containing the reason that this transfer
+was rejected.
+
+#### Response
+
+A `Reject` request must come from the receiver of the referenced transfer,
+not the sender.
+
+A `Reject` request is successful if `transferId` references an existing
+transfer in the `prepared` state.
+
+A `Reject` request is also successful if `transferId` references an existing
+transfer in the `rejected` state, and this packet matches the one that
+rejected this transfer, including side protocols but not including requestId.
+
+- A `Response` is returned if the request is successful. This indicates that
+  the balance changes of the referenced transfer have been rolled back.
```

Comment:
I still think the current implementation with `Ack` is nicer; it feels weird to call it a "response" if you have nothing to say.